core/coreunix/add.go

```diff
@@ -22,8 +22,12 @@ import (
 
 var log = eventlog.Logger("coreunix")
 
-// Add builds a merkledag from the a reader, pinning all objects to the local
-// datastore. Returns a key representing the root node.
+// DEPRECATED: Add builds a merkledag from the a reader, pinning all
+// objects to the local datastore. Returns a key representing the root
+// node.
+//
+// This function is deprecated and will be removed in 0.4.0.  You
+// should use shell/fsnode's AddFromReader instead.
```

Comment:
@wking sorry--  i don't think we're communicating...  i don't want this deprecated from core. i don't see the why it needs to be yet.

I'm not 100% sure about this right now, but it seems to me that all the things that the shell can do, will likely be in core as well. (the shell is mostly just a wrapper). Perhaps I should spend time charting out what shell and core should look like, i can do that next sprint.

the core is where the bulk of the operations will remain, the shell is just a very simplified interface that makes it easier for users to use ipfs nodes that are either embedded or over an rpc transport.

okay, i haven't made things very clear with all this. we should spec out both the shell and the core should be -- on their own as interface specs -- before deciding what to deprecate, or actually deprecating, from the current interfaces. trying to decide one thing goes here and not there without the full picture -- and with clearly different mental models -- is going to cause undue decision conflicts. i'll make it a priority to get that out next week.